Improve docs for receiving syslog over TLS (#210)
pganalyze/pganalyze-docs: log-insights/setup/syslog-server.mdx

---

If your database logs are available via syslog, you can use the [`db_log_syslog_server`](/docs/collector/settings#self-managed-servers)
setting to have the collector listen for syslog messages from a remote server.
When this is specified, the collector will run the syslog server behind the

> scene on the specified port to receive syslog messages.

Reviewer: ```suggestion
scenes on the specified port to receive syslog messages.
```

This should be a local address like "0.0.0.0:32514" and it's recommended to use
an unprivileged port number (1024 and up) to avoid running the syslog server as
root.

Note that you need to use protocol RFC5424, with a TCP connection. When not

> using TLS, it is recommended to only use this integration over private networks.

Reviewer: I feel like this waters down this warning a bit. If you're using TLS but _not_ using client certs, someone could still be sending bogus logs to your collector and you won't be able to tell. Now, the likelihood of this happening is pretty small, and the damage they can do is pretty limited (DoS your collector? cause bogus information to show up in your Log Insights page?), but I think I'd still rather make it clear what the situation is and allow the user to decide whether they want to take that chance.

Author: Yeah fair point. I'll tweak to cover that

Author: Uhmm while trying to tweak this, I started wondering about our choice `VerifyClientCertIfGiven` https://pkg.go.dev/crypto/tls#ClientAuthType

You're right that the using TLS won't be totally secure, in terms of anyone can send out logs to the collector (still secure in terms of eavesdropping and tampering). When someone actually wants to ensure the mTLS (aka prevent from any random folks sending out logs), they can't really do this with `VerifyClientCertIfGiven`, because we only verify when the client cert is given, so bad person can still send out logs without the client cert and the collector will just receive it.
I think `VerifyClientCertIfGiven` is still valid as a default behavior, though we might consider supporting additional options if there's a request for such a feature.

Author: Update, maybe I'd love to have one more look this part (no rush).

Author: Gonna merge as is but feel free to leave any comments if you have, happy to create a follow-up PR

Reviewer: The revised text looks good. Good point on `VerifyClientCertIfGiven`. I think you're right; we can address that if it comes up.

### Receiving syslog over TLS

You can also receive logs over TLS by specifying the path of a certificate [`db_log_syslog_server_cert_file`](/docs/collector/settings#self-managed-servers)
and a secret key [`db_log_syslog_server_key_file`](/docs/collector/settings#self-managed-servers).
Alternatively, you can also set the actual contents of a certificate with [`db_log_syslog_server_cert_contents`](/docs/collector/settings#self-managed-servers),
or a secret key with [`db_log_syslog_server_key_contents`](/docs/collector/settings#self-managed-servers).
Please note that if the `db_log_syslog_server_cert_file` setting is provided,

> the connection needs to be made as TLS and the connection attempt without TLS

Reviewer: ```suggestion
the connection needs to be made as TLS and connection attempts without TLS
```

will be rejected.

> This certificate is for the syslog server running on the collector. You have the

Reviewer: ```suggestion
This certificate is for the syslog server running as part of the collector. You have the
```

option to use an existing certificate or generate a new self-signed certificate
specifically for this purpose. In cases where a self-signed certificate or a
certificate from an untrusted authority is used, it is also essential to specify
the path to the TLS Certificate Authority (CA) certificate(s) with the [`db_log_syslog_server_ca_file`](/docs/collector/settings#self-managed-servers)

> setting. This CA certificate is also required to provide to the client that is
> sending out logs to the syslog server, such as rsyslog.

Reviewer: ```suggestion
setting. This CA certificate must also be provided to the client that is
sending out logs to the syslog server, such as rsyslog.
```

With these settings, logs from the client (like rsyslog) to the collector syslog
server will be sent over TLS.

You can also enable mutual TLS to ensure that the client side is also verified.
The client side can provide a certificate for this, and if this certificate is
a self-signed certificate or a certificate from an untrusted authority, the

> collector syslog server needs to know about the CA certificate of it. You can

Reviewer: ```suggestion
collector syslog server needs to know about the CA certificate for it. You can
```

specify the CA certificate of the client side using the [`db_log_syslog_server_client_ca_file`](/docs/collector/settings#self-managed-servers)

> setting.

Reviewer: ```suggestion
or [`db_log_syslog_server_client_ca_contents`](/docs/collector/settings#self-managed-servers) setting.
```
